### DISCUSSION TITLE
Detailed deployment tutorial?help me !!

### DISCUSSION THREAD

**Discussion by coldpub** [spencerwooo/onedrive-vercel-index#193] [Q&A] [answered]
Is there a detailed deployment tutorial? help me! !

**Comment by myl7** [comment]
See https://github.com/spencerwooo/onedrive-cf-index README

**Reply by coldpub** [reply]
I can't get refresh_token

**Reply by coldpub** [reply]
When I enter npx @beetcb/ms-graph-cli, I can't resolve the error node:internal/modules/cjs/loader:936 throw err; ^ Error: Cannot find module 'flatstr'

**Comment by coldpub** [comment]
{
"error": "invalid_request",
"error_description": "AADSTS9002326: Cross-origin token redemption is permitted only for the 'Single-Page Application' client-type.\r\nTrace ID: 4bb19cfa-1fa0-4ec3-a886-d280873bc200\r\nCorrelation ID: b07661d2-9a27-49ff-9839-9d719af6c9aa\r\nTimestamp: 2021-12-26 03:57:44Z",
"error_codes": [
9002326
],
"timestamp": "2021-12-26 03:57:44Z",
"trace_id": "4bb19cfa-1fa0-4ec3-a886-d280873bc200",
"correlation_id": "b07661d2-9a27-49ff-9839-9d719af6c9aa",
"error_uri": "https://login.microsoftonline.com/error?code=9002326"
}
点我阅读

**Comment by spencerwooo** [comment]
are you following the documentation in the [previous project](https://github.com/spencerwooo/onedrive-cf-index#generating-onedrive-api-tokens)? judging from the errors ~~i should believe you didn't choose `Redirect URI (optional)` to `Web` in the first step when you are registering the application~~ never mind its not that.

you know what, please just wait until i code a website for helping you obtain the tokens, i should have some time next week. cheers :)

**Reply by coldpub** [reply]
Okay, thanks!! Come on!!!

**Comment by coldpub** [comment]
yes

I set up redirects

**Comment by spencerwooo** [comment, marked as answer]
ok, new version released with a simplified deploy process, try it out ;)
